src/k8s/pkg/k8sd/app/hooks_bootstrap.go

```diff
@@ -325,62 +325,105 @@ func onBootstrapControlPlane(s *state.State, initConfig map[string]string) error
 	}
 
 	if cfg.Network.Enabled != nil {
-		err := component.ReconcileNetworkComponent(s.Context, snap, vals.Pointer(false), cfg.Network.Enabled, cfg)
-		if err != nil {
+		network := &component.Network{
+			PodCIDR: cfg.Network.PodCIDR,
+		}
```

Comment:
not a huge fan of this, as there are no clear semantics as to which fields are inputs and which are outputs. I'd recommend reverting these changes